docs/root/configuration/http/http_conn_man/vhds.rst

```diff
@@ -0,0 +1,115 @@
+in the mesh. This causes scaling issues as the size of the cluster grows. The
+majority of this complexity can be found in the virtual host configurations, of
+which most are not needed by any individual proxy. 
+
+In order to fix this issue, the Virtual Host Discovery Service (VHDS) protocol
+uses the delta xDS protocol to allow a route configuration to be subscribed to
+and the necessary virtual hosts to be requested as needed. Instead of sending
+all virtual hosts with a route config, using VHDS will allow an Envoy instance
+to subscribe and unsubscribe from a list of virtual hosts stored internally in
+the xDS management server. The xDS management server will monitor this list and
+use it to filter the configuration sent to an individual Envoy instance to only
+contain the subscribed virtual hosts.
+
+Virtual Host Resource Naming Convention
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Virtual hosts in VHDS are identified by a combination of the name of the route
+configuration to which the virtual host belong as well as the host HTTP "host"
+header (authority for HTTP2) entry. Resources should be named as follows:
+
+<route configuration name>/<host entry>
+
+Note that matching should be done from right to left since a host entry cannot
+contain slashes while a route configuration name can. 
+
+Subscribing to Resources
+^^^^^^^^^^^^^^^^^^^^^^^^
+VHDS allows resources to be :ref:`subscribed <xds_protocol_delta_subscribe>` to
+using a :ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+with the 
+:ref:`type_url <envoy_api_field_DeltaDiscoveryRequest.type_url>` set to 
+`type.googleapis.com/envoy.api.v2.route.VirtualHost` 
+and :ref:`resource_names_subscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_subscribe>` 
+set to a list of virtual host resource names for which it would like
+configuration. 
+
+If a route for the contents of a host/authority header cannot be resolved,
+the active stream is paused while a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>` is sent. 
+When a :ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>` is
+received where one of the :ref:`aliases <envoy_api_field_resource.aliases>` in
+the response exactly matches the
+:ref:`resource_names_subscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_subscribe>`
+entry from the 
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`,
+the route configuration is updated, the stream is resumed, and processing of the
+filter chain continues.
+
+Each of the virtual hosts contained in the
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+resources field will be added to the route configuration maintained by Envoy.
+If Envoy's route configuration already contains a given virtual host, it will
+be overwritten but only if the updated virtual host is different from its
+current state. The management server needs to keep track of virtual hosts
+known to Envoy.
```

Comment:
Where did we end up here? I thought we were going in the direction of making the static/dynamic virtual hosts orthogonal?

Reply:
Good point -- I missed this one earlier.  I think these last two sentences should be removed.

Reply:
They are, as illustrated a couple of paragraphs below:

Updates to virtual hosts occur in two ways. If a virtual host was originally
sent over RDS, then the virtual host should be updated over RDS. If a virtual
host was subscribed to over VHDS, then updates will take place over VHDS.
 
That being said, I think moving that paragraph above might clarify this a bit.

Reply:
Cleaned up the content.

Reply:
This looks good to me now.